File: python/datashaper/datashaper/engine/verbs/drop.py
#
# Copyright (c) Microsoft. All rights reserved.
# Licensed under the MIT license. See LICENSE file in the project.
#
"""Drop verb implementation."""

from typing import Any

from datashaper.engine.verbs.verbs_mapping import verb
from datashaper.table_store.types import VerbResult, create_verb_result
from datashaper.verbs import drop

from .verb_input import VerbInput


@verb(name="drop")
def drop_verb(
    input: VerbInput,
    **kwargs: Any,
) -> VerbResult:
    """Drop verb implementation."""
    result = drop(input.get_input(), **kwargs)
    return create_verb_result(result)

File: javascript/workflow/src/verbs/drop.ts
/*!
 * Copyright (c) Microsoft. All rights reserved.
 * Licensed under the MIT license. See LICENSE file in the project.
 */
import type { DropArgs } from '@datashaper/schema'
import { not } from 'arquero'

import type { ColumnTableStep } from './util/factories.js'
import { stepVerbFactory } from './util/factories.js'

export const dropStep: ColumnTableStep<DropArgs> = (
	input,
	{ columns = [] },
) => {
	return input.select(not(columns))
}

export const drop = stepVerbFactory(dropStep)
